pkg/daemon/daemon.go

```diff
@@ -2589,6 +2589,21 @@ func (dn *Daemon) triggerUpdateWithMachineConfig(currentConfig, desiredConfig *m
 	return dn.update(currentConfig, desiredConfig, skipCertificateWrite)
 }
 
+func (dn *Daemon) inPlaceApplyChanges(oldConfig, newConfig *mcfgv1.MachineConfig, currentImage string) error {
+	err := dn.update(oldConfig, newConfig, true)
```

Comment:
issue: Call `canonicalizeMachineConfigImage()` on both the `oldConfig` and the `newConfig` here, otherwise the OCL-provided OS image will not be applied.